### ISSUE TITLE
Occassional corruption of request payload

### ISSUE THREAD

**Issue by dodysw2** [buger/goreplay#952]
I think there's a race condition in goreplay that causes corruption of captured requests. My particular setup to reproduce this is by tcpdump incoming http traffic to a pcap file, and run gor using that pcap file as input, and output to an http end point else where, and made it `-verbose 1` to show errors.

Occassionally, and always different on every invocation, got `malformed chunked encoding` errors, or `http: unexpected EOF reading trailer`, or sometimes, no error. On the output http end point I can confirm that the request sent by gor was corrupted.

As debugging this, I found that capturing http traffic with chunked encoding is the easiest way to detect this error, as gor tripped on its own code when trying to forward the request (there's some sort of validation on golang httpclient library). However, I can confirmed (by also outputting to file) that the corruption happend not specifically on the http output routine, as the output file also contains the error.

I have also validate that the pcap file do not have issues, by manually verifying using wireshark on the particular errors triggered on gor.

If you need me to try out something to remove possibilities of a bug in gor, please let me know. Thanks!

**Comment by buger** [comment]
Hi! 

If you can provide some easy way to replicate the issue will be great!

The first step is to increase timeout, maybe your server or networking is a bit slow, and you need to give it a chance to wait for all packets. Smth like this: `--input-raw-expire 1s`. 


I have a feeling it is this part of the code:
https://github.com/buger/goreplay/blob/master/tcp/tcp_message.go#L318-L327

I just added a fix for this for upcoming 1.3.0 will be great if you can test it.
https://github.com/buger/goreplay/releases/tag/v1.3.0_RC6

It will skip messages HTTP messages which was not received till timeout hit, or messages which are not fully complete. Can be enabled back using `--input-raw-allow-incomplete`

**Comment by dodysw2** [comment]
Yes, I can forward you the pcap file and parameters used via the email.

The RC6 binary isn't running in my system with `panic: ./gor flag redefined: cpuprofile` error. I removed the duplicate flags and also fix the `package main` inside af_packet_linux.go that prevented the build also and tried again, but the issue persist.

`--input-raw-expire 1s` isn't an issue I guess because the problem also occured on pcap file, and the default is already 2s btw. 

In parallel I also tried (on the code before RC6) tracing the goroutines and found that when the pcap reader finished the reading (no more packets) it returned EOF / ErrorStop error to a quit channel which seems to prematurely terminates the read loop (not all of the packets/messages have been read) which I guess likely because my pcap file is very small (only 24 packets).  

```
	case <-i.quit:
		return nil, ErrorStopped
	case msgTCP = <-i.messageParser.Messages():
		msg.Data = msgTCP.Data()
	}
```
No idea if it has anything to do with the corruption when not using pcap file.

**Comment by dodysw2** [comment]
Just to update -- still in progress localizing using tcpdump pcap file, which part consistently triggers the issue. I found using small sample rarely triggers it, but a larger sample is quite challenging to sanitize/scrube the data. Will update soon on next few days.
